src/liballoc/char.rs

```diff
@@ -0,0 +1,26 @@
+/// Implements the `+` operator for concatenating two `char`s together.
+///
+/// This operation results into a new `String` being allocated with copies of the two `char`s in
+/// the respective order.
+///
+/// # Examples
+///
+/// ```
+/// let a = '🎈';
+/// let b = '🎉';
+/// let c = a + b;
+/// // `c` is a newly allocated `String`
```

Comment:
```suggestion
/// assert_eq!(c, "🎈🎉");
```

Examples are automatically tested by rustdoc.